modular_skyrat/modules/armaments/code/armament_entries.dm

```diff
@@ -0,0 +1,108 @@
+				armament_dataset[spawned_armament_entry.category][CATEGORY_ENTRY][ARMAMENT_SUBCATEGORY_NONE] += spawned_armament_entry
+			// Set the category item limit.
+			armament_dataset[spawned_armament_entry.category][CATEGORY_LIMIT] = spawned_armament_entry.category_item_limit
+		else
+			// Because of how the UI system works, categories cannot exist with nothing in them, so we
+			// only set the OTHER category if something can go inside it! This seems like a copy paste job, but it needs to be here.
+			if(!(ARMAMENT_CATEGORY_STANDARD in armament_dataset))
+				armament_dataset[ARMAMENT_CATEGORY_STANDARD] = list(CATEGORY_ENTRY, CATEGORY_LIMIT)
+				armament_dataset[ARMAMENT_CATEGORY_STANDARD][CATEGORY_LIMIT] = ARMAMENT_CATEGORY_STANDARD_LIMIT
+			// We don't have home :( add us to the other category.
+			if(spawned_armament_entry.subcategory)
+				// Check to see if we've already made the subcategory.
+				if(!(spawned_armament_entry.subcategory in armament_dataset[ARMAMENT_CATEGORY_STANDARD][CATEGORY_ENTRY]))
+					armament_dataset[ARMAMENT_CATEGORY_STANDARD][CATEGORY_ENTRY][spawned_armament_entry.subcategory] = list()
+				// Finally, we add the entry into the list.
+				armament_dataset[ARMAMENT_CATEGORY_STANDARD][CATEGORY_ENTRY][spawned_armament_entry.subcategory] += spawned_armament_entry
+			else
+				// Unset subcategories default to the NONE category.
+				if(!(ARMAMENT_SUBCATEGORY_NONE in armament_dataset[ARMAMENT_CATEGORY_STANDARD][CATEGORY_ENTRY]))
+					armament_dataset[ARMAMENT_CATEGORY_STANDARD][CATEGORY_ENTRY][ARMAMENT_SUBCATEGORY_NONE] = list()
+				armament_dataset[ARMAMENT_CATEGORY_STANDARD][CATEGORY_ENTRY][ARMAMENT_SUBCATEGORY_NONE] += spawned_armament_entry
+	return armament_dataset
+
+//////////////////////
+// ARMAMENT ENTRIES //
+//////////////////////
+/datum/armament_entry
+	/// The name of the equipment used in the listing, if not set, it will use the items name.
+	var/name
+	/// The description of the equipment used in the listing, if not set, it will use the items description.
+	var/description
+	/// The item path that we refer to when equipping. If left empty, it will be considered abstract.
+	var/obj/item_type
+	/// Category of the item. This is used to group items together in the UI.
+	var/category = ARMAMENT_CATEGORY_STANDARD
+	/// This is an abstract variable, only set this for base category types. It should not be overriden by subtypes. Set to 0 for infinite.
+	var/category_item_limit = 0
+	/// Our subcategory, where the item will be listed.
+	var/subcategory = ARMAMENT_SUBCATEGORY_NONE
+	/// The points cost of this item.
+	var/cost = 0
+	/// Defines what slot we will try to equip this item to.
+	var/slot_to_equip = ITEM_SLOT_HANDS
+	/// Our cached image.
+	var/cached_base64
+	/// The maximum amount of this item that can be equipped.
+	var/max_purchase = 1
+
+/datum/armament_entry/proc/create_icon_cache() // TODO: Make this use overlays.
```

Comment:
will this TODO be done? as-is pistol slides and stuff are invisible

Reply:
No not in this PR, probably in the gundealer PR where it expands on this.